src/Engine/Rendering/ForwardRenderer.cpp

```diff
@@ -371,106 +453,50 @@ void ForwardRenderer::renderInternal( const Data::ViewingParameters& renderData
                 using polymesh = Ra::Engine::Data::IndexedGeometry<Ra::Core::Geometry::PolyMesh>;
 
                 auto displayable = ro->getMesh();
                 auto tm          = std::dynamic_pointer_cast<trimesh>( displayable );
                 auto tp          = std::dynamic_pointer_cast<polymesh>( displayable );
-                //                Ra::Core::Geometry::TopologicalMesh topo;
-                bool drawable = false;
-                Core::Geometry::LineMesh lines;
-                Core::Geometry::LineMesh::IndexContainerType indices;
+
                 if ( tm )
                 {
-                    drawable = tm->getRenderMode() ==
-                               Data::AttribArrayDisplayable::MeshRenderMode::RM_TRIANGLES;
-                    if ( drawable )
-                    {
-                        lines.setVertices( tm->getCoreGeometry().vertices() );
-                        for ( const auto& index : tm->getCoreGeometry().getIndices() )
-                        {
-                            for ( unsigned int i = 0; i < 3; ++i )
-                            {
-                                int i1 = index[i];
-                                int i2 = index[( i + 1 ) % 3];
-                                if ( i1 > i2 ) std::swap( i1, i2 );
-                                indices.emplace_back( i1, i2 );
-                            }
-                        }
-                    }
+                    if ( tm->getRenderMode() ==
+                         Data::AttribArrayDisplayable::MeshRenderMode::RM_TRIANGLES )
+                    { setupLineMesh( disp, tm->getCoreGeometry() ); }
                 }
                 if ( tp )
                 {
-                    drawable = tp->getRenderMode() ==
-                               Data::AttribArrayDisplayable::MeshRenderMode::RM_TRIANGLES;
-
-                    if ( drawable )
-                    {
-                        drawable = tp->getRenderMode() ==
-                                   Data::AttribArrayDisplayable::MeshRenderMode::RM_TRIANGLES;
-                        lines.setVertices( tp->getCoreGeometry().vertices() );
-                        for ( const auto& index : tp->getCoreGeometry().getIndices() )
-                        {
-                            auto s = index.size();
-                            for ( unsigned int i = 0; i < s; ++i )
-                            {
-                                int i1 = index[i];
-                                int i2 = index[( i + 1 ) % s];
-                                if ( i1 > i2 ) std::swap( i1, i2 );
-                                indices.emplace_back( i1, i2 );
-                            }
-                        }
-                    }
+                    if ( tp->getRenderMode() ==
+                         Data::AttribArrayDisplayable::MeshRenderMode::RM_TRIANGLES )
+                    { setupLineMesh( disp, tp->getCoreGeometry() ); }
```

Comment:
There is here unneeded code duplication (from line 473 to 483), could be replaced by 
```.cpp
                auto processLineMesh = []( auto m, std::shared_ptr<Data::LineMesh>& disp ) {
                    if ( m->getRenderMode() ==
                         Data::AttribArrayDisplayable::MeshRenderMode::RM_TRIANGLES )
                    { setupLineMesh( disp, m->getCoreGeometry() ); }
                };
                if ( tm ) { processLineMesh( tm, disp ); }
                if ( tp ) { processLineMesh( tp, disp );
```

Reply:
done